### ISSUE TITLE
Formatting of find_package with multiple REQUIRED modules inconsistent

### ISSUE THREAD

**Issue by PatTheMav** [BlankSpruce/gersemi#20]
Take the following CMake line:

```
find_package(FFmpeg ${ffmpeg_version} REQUIRED avcodec avfilter avdevice avutil swscale avformat swresample)
```

I would have expected the formatting to produce code like this:
```
find_package(
  FFmpeg
  ${ffmpeg_version}
  REQUIRED
    avcodec
    avfilter
    avdevice
    avutil
    swscale
    avformat
    swresample
)
```

`gersemi` 0.12.0 is _almost_ there, but doesn't indent the `REQUIRED` modules.

**Comment by BlankSpruce** [comment]
It's because the modules (cmake doc calls them components) don't belong to `REQUIRED`. 

Your example fits this part of signature https://cmake.org/cmake/help/latest/command/find_package.html#full-signature
```
find_package(
    <PackageName>
    [version]
    [REQUIRED]
    [[COMPONENTS] [components...]]
)
```

Even if I made a heuristic exception like "treat list of modules after `REQUIRED` as if they belong to that keyword then this heuristic falls apart when either `REQUIRED` is not present or it's placed after the components (I'm not 100% sure it'd work but for many commands keywords don't need to be in order presented in signature). I can only recommend using the explicit `COMPONENTS` keyword.

**Comment by PatTheMav** [comment]
At least as far as CMake's documentation is concerned, it's correct:

> As a form of shorthand, if the `REQUIRED` option is present, the `COMPONENTS` keyword can be omitted and the required components can be listed directly after `REQUIRED`.

The signature even spells this out: `[[COMPONENTS] [components...]]` means (to me at least) that `COMPONENTS` itself is an ignored part of the components list following `REQUIRED`, so it should be the other way around - treat everything after `REQUIRED` (but which is not one of the other tokens from the signature) as a component and ignore the `COMPONENTS` keyword.

**Comment by BlankSpruce** [comment]
Nice finding, I must have missed that detail. That's one of the pain points of doing this stuff well, signatures aren't enough sometimes.

I propose then that the formatting will be as following (just expanded forms):
```
find_package(
    FOO
    REQUIRED
        bar
        baz
)

find_package(
    FOO
    COMPONENTS
        bar
        baz
)

find_package(
    FOO
    REQUIRED
    COMPONENTS
        bar
        baz
)
```

I won't merge these two keywords into one line `REQUIRED COMPONENTS` since `REQUIRED` has specific, option-like, semantics:
>  If REQUIRED is specified and the package is not found a fatal error is generated and the configure step stops executing. 

It'll be fixed in 0.12.1.

**Comment by PatTheMav** [comment]
> Nice finding, I must have missed that detail. That's one of the pain points of doing this stuff well, signatures aren't enough sometimes.
> 
> I propose then that the formatting will be as following (just expanded forms):
> 
> ```
> find_package(
>     FOO
>     REQUIRED
>         bar
>         baz
> )
> 
> find_package(
>     FOO
>     COMPONENTS
>         bar
>         baz
> )
> 
> find_package(
>     FOO
>     REQUIRED
>     COMPONENTS
>         bar
>         baz
> )
> ```
> 
> I won't merge these two keywords into one line `REQUIRED COMPONENTS` since `REQUIRED` has specific, option-like, semantics:
> 
> > If REQUIRED is specified and the package is not found a fatal error is generated and the configure step stops executing.
> 
> It'll be fixed in 0.12.1.

No worries, you're still at major version 0 after all. 😄

Proposal looks fine to me.

**Comment by BlankSpruce** [comment]
That zero major version is my "out of jail" card. :)

0.12.1 just got released.
